src/client/create-http-client.ts

```diff
@@ -1,28 +1,22 @@
-import axios from 'axios';
+import axios, {AxiosRequestConfig} from 'axios';
 import * as pathToRegExp from 'path-to-regexp';
 import {ParamNames, Paths, RequestBody, ResponseBody} from '../util';
 import {HttpSchema} from '../shared';
 
 
 /** Returns a strongly typed object for making requests to a remote HTTP server that implements the given `schema`. */
-export function createHttpClient<S extends HttpSchema>(schema: S, options?: Partial<HttpClientOptions>): HttpClient<S> {
+export function createHttpClient<S extends HttpSchema>(schema: S, options?: AxiosRequestConfig): HttpClient<S> {
 
-    // Create an axios client for making actual HTTP requests. Initialise it with the relevent given options, if any.
-    const axiosClient = axios.create({
-        baseURL: options?.baseURL,
-        timeout: options?.timeout ?? 0,
-        withCredentials: options?.withCredentials ?? false,
-    });
+    // Create an axios client for making actual HTTP requests. Initialise it with the relevant given options, if any.
+    const axiosClient = axios.create(options);
```

Comment:
There's no need to limit the options here, we can just use the Axios type and pass the options param thru, simplifies things a bit and allows users to use Axios features.